### PR TITLE
chore(overlay, tooltip): v0 directive for Overlay v2 testing

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Directives, demos, and tests for lazily rendering Overlay content.

This prepares for an initial pre-release for use in major consuming clients for testing purposes.

*EVERYTHING HERE IS EXPERIMENTAL!*

We're looking at pre-releasing this to `latest` and then revisit after some testing for documentation/announcement.

_Known issues:_
- directives do not orchestrate with each other, so using more than one on the same trigger is not currently suggested
- lazily populating an `<sp-overlay>` should be _roughly_ as performant as the directive, but not currently sure why there's a performance delta between the two (this lead to some gaming of the class structure that could technically be reverted)
- I'm not super happy with this argument shape or the defaults it represents:
  ```
        triggerInteraction: 'hover',
        overlayOptions: {
            placement: 'top-start',
            type: 'auto',
            offset: 0,
        },
  ```
  further pre-release experimentation will likely smooth that out.
- there are additional DEV MODE warnings included for the experimental nature of this code
- there's not a lot of tests of the directive

## Related issue(s)
- refs #3549

## How has this been tested?
-   [ ] _Test case 1_
    1. Go [here](https://overlay-directive-v2--spectrum-web-components.netlify.app/storybook/?path=/story/overlay-directive--default)
    2. See the tooltip and the popover related to the button

## Types of changes
-   [x] New feature (non-breaking change which adds functionality)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: packages/overlay/src/overlay-trigger-directive.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+        render(willRemoveSlottable ? undefined : this.template(), this.overlay);
+        if (willRemoveSlottable) {
+            this.overlay.remove();
+        } else {
+            Overlay.applyOptions(this.overlay, {
+                ...this.options,
+                trigger: this.target,
+            });
+            this.target.insertAdjacentElement('afterend', this.overlay);
```
Comment (hunterloftis) [inline packages/overlay/src/overlay-trigger-directive.ts:117]: Should we make the insertion point configurable or is after the target always ok?

File: packages/overlay/stories/overlay-directive.stories.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        <sp-button
+            variant="primary"
+            ${tooltip(
+                () =>
+                    html`
+                        Click to open a popover.
+                    `
+            )}
+            ${trigger(
+                () => html`
```
Comment (hunterloftis) [inline packages/overlay/stories/overlay-directive.stories.ts:154]: I know it's just a story, but I wonder what you think of naming the various parts as render functions just for readability:

```
const renderPopover = () => html`
  ...
`;

<sp-button ${trigger(renderPopover)}>
```

File: packages/tooltip/src/tooltip-directive.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    options?: Partial<OverlayTriggerOptions & { variant: string }>
+): ReturnType<typeof trigger> {
+    return trigger(() => {
+        import('@spectrum-web-components/tooltip/sp-tooltip.js');
+        return html`
+            <sp-tooltip variant=${ifDefined(options?.variant)}>
+                ${tooltipContent()}
+            </sp-tooltip>
+        `;
+    }, options);
```
Comment (hunterloftis) [inline packages/tooltip/src/tooltip-directive.ts:30]: 🔥 🔥 🔥

**Review by hunterloftis — DISMISSED** [review]
Very interested to see what impact this has.

I also suspect that adopting these directives will expose places where apps are inadvertently relying on the DOM to maintain state.

File: packages/overlay/src/overlay-trigger-directive.ts (hunk 1 above)
Comment (Westbrook) [reply packages/overlay/src/overlay-trigger-directive.ts]: Added.

**Comment by beeduul** [comment]
> I also suspect that adopting these directives will expose places where apps are inadvertently relying on the DOM to maintain state.

that could be really interesting if so.  and a statement that doing things the "right way" benefits performance AND robustness/correctness.

**Review by hunterloftis — APPROVED** [review]

**Pull request merged**